pkg/subctl/cmd/root.go

```diff
@@ -234,3 +240,25 @@ var nodeLabelBackoff wait.Backoff = wait.Backoff{
 	Factor:   1.2,
 	Jitter:   1,
 }
+
+func checkVersionMismatch(cmd *cobra.Command, args []string) error {
+	config, err := getRestConfig(kubeConfig, kubeContext)
+	exitOnError("The provided kubeconfig is invalid", err)
+
+	clientSet, err := submarinerclientset.NewForConfig(config)
+	exitOnError("Unable to get the Submariner config", err)
+
+	submariner, _ := clientSet.SubmarinerV1alpha1().Submariners(OperatorNamespace).Get(submarinercr.SubmarinerName, metav1.GetOptions{})
+
+	if submariner != nil && submariner.Spec.Version != "" {
```

Comment:
We should be consistent and fail on error as we do above:

```suggestion
	submariner, err := clientSet.SubmarinerV1alpha1().Submariners(OperatorNamespace).Get(submarinercr.SubmarinerName, metav1.GetOptions{})
	
        if apiErrors.IsNotFound(err) {
	    return nil
         }
	
         exitOnError("Error retrieving the Submariner resource", err)

	if submariner.Spec.Version != "" {
```

Note: `import apiErrors "k8s.io/apimachinery/pkg/api/errors"`